src/main/java/org/broadinstitute/hellbender/engine/filters/AmbiguousBaseReadFilter.java

```diff
@@ -15,30 +15,29 @@
     private static final long serialVersionUID = 1L;
 
     @Argument(doc = "Threshold fraction of ambiguous bases",
-            fullName = "ambigFilterFrac",
+            fullName = "maxAmbiguousBases",
             optional = true)
-    public double nFrac = 0.05;
+    public int maxAmbiguousBases = 0;
 
     public AmbiguousBaseReadFilter() {
     }
 
-    public AmbiguousBaseReadFilter(final double n_frac) {
-        this.nFrac = n_frac;
+    public AmbiguousBaseReadFilter(final int maxAmbiguousBases) {
+        this.maxAmbiguousBases = maxAmbiguousBases;
     }
 
     //Filters out reads with more than a threshold number of N's
     @Override
     public boolean test(final GATKRead read) {
-        final int N_max = (int) (read.getLength() * nFrac);
```

Comment:
Just added your first suggestion to use another variable for maxAmbiguousBases that is null by default but can be set using a different constructor.